src/GameController.java

```diff
@@ -102,7 +95,12 @@ private boolean isMovementValid(Player player, Direction direction) {
         final int y = player.getYPos();
         boolean result = false;
         try {
-            // TODO: Explain lock.tryLock, lock.unlock, and how they interact with the game board to prevent collisions
+            // The game board is a thread-locked object which only one player thread can modify at a time.
+            // Upon attempting to check if movement is valid, attempt to acquire a lock on the board.
+            // If it succeeds, i.e. the board is not currently held by another server-side player thread:
+            // 1) check for out-of-range coordinates to make sure a move doesn't escape the board
+            // 2) check if a tile is 'TRUE' in the occupancy map; if so, it is occupied by another player.
```

Comment:
Other way around

TRUE means it is valid
FALSE means someone is in that position, thus invalid

Reply:
Thanks! Had it right in the report but wrong here